app/code/Magento/Bundle/Model/Product/Type.php

```diff
@@ -536,15 +536,17 @@ public function updateQtyOption($options, \Magento\Framework\DataObject $option,
 
         foreach ($selections as $selection) {
             if ($selection->getProductId() == $optionProduct->getId()) {
-                foreach ($options as &$option) {
-                    if ($option->getCode() == 'selection_qty_' . $selection->getSelectionId()) {
+                foreach ($options as &$quoteItemOption) {
```

Comment:
Why do we need foreach by reference here at all? It is an object, please just remove ampersand and `unset`.